src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
```

Comment:
The code can be optimized to avoid repeating the same filter or finding operations multiple times. For example, in calculateAverageAgeDiff, instead of finding the mother's object for each child in the loop, we can pre-process the array to create a map from the mother's name to her object and use it to look up the mother's object directly.